nemo/src/execution/execution_engine/tracing/node_query/util.rs

```diff
@@ -362,8 +384,45 @@ pub(super) async fn ignore_discarded_columns_base(
         .map(|(_, &result_id)| result_id)
 }
 
+/// Filters the input table by the elements in the query table,
+/// by joining input and query table.
+///
+/// Returns the id of the input table if the query table is `None`.
+///
+/// Return None if the execution of the plan failed.
+pub(super) async fn join_query_node(
+    database: &mut DatabaseInstance,
+    arity: usize,
+    input_table: PermanentTableId,
+    query_table: Option<PermanentTableId>,
+) -> Option<PermanentTableId> {
+    let query_table = if let Some(id) = query_table {
+        id
+    } else {
+        return Some(input_table);
+    };
+
+    let mut plan = ExecutionPlan::default();
+    let markers = OperationTable::new_unique(arity);
+
+    let node_input = plan.fetch_table(markers.clone(), input_table);
+    let node_query = plan.fetch_table(markers.clone(), query_table);
+
+    let node_join = plan.join(markers, vec![node_input, node_query]);
+
+    plan.write_permanent(node_join, "join", "join");
+
+    database
+        .execute_plan(plan)
+        .await
+        .expect("error while executing plan")
```

Comment:
The `expect` string should not say what went wrong but imo it should state the assumption *why this is expected to work*!